torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -1085,8 +1129,13 @@ bool checkCanSchedule(
   // since for all current use cases
   //  it has to pass all the compile time checks to create a data cache for this
   //  fusion.
-  if (!data_cache && !SchedulerType::canScheduleCompileTime(fusion)) {
-    return false;
+  if (!data_cache){
+    if(!isConnectedFusionGraph(fusion)){
```

Comment:
So independent data flow is disabled for all schedulers? I thought it only breaks PW scheduler and we would still want horizontal fusion for others. I could be wrong though.

Reply:
My understanding is that independent data flow would not work smoothly on other heuristics either. If even root domain map doesn't have useful info on the un-connected components, it's hard to have a sound analysis on the tensor dimensions and parallelization strategy.

Reply:
But I guess these cases wouldn't be sent to reduction or persistent schedulers either way, since root domain map was used in the canSchedule check.

Reply:
Got ya.